SourceCode/GPS/Classes/CISOBUS.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AgOpenGPS
+{
+    public class CISOBUS
+    {
+        private readonly FormGPS mf;
+
+        private bool wasAlive;
+        private long timestamp;
+
+        private bool sectionControlEnabled;
+        private List<bool> actualSectionStates = new List<bool>();
+
+        public CISOBUS(FormGPS _f)
+        {
+            //constructor
+            mf = _f;
+        }
+
+        public bool IsSectionOn(int section)
+        {
+            if (section < actualSectionStates.Count)
+            {
+                return actualSectionStates[section];
+            }
+            return false;
+        }
+
+        public void RequestSectionControlEnabled(bool enabled)
+        {
+            // Send the request
+            byte[] data = new byte[7];
+            data[0] = 0x80; // standard AIO header
+            data[1] = 0x81; // PGN header
+            data[2] = 0x7F; // SRC address
+            data[3] = 0xF1; // PGN
+            data[4] = 0x02; // Length
+            data[5] = (byte)(enabled ? 0x01 : 0x00); // Section control enabled request
+            mf.SendPgnToLoop(data);
+        }
+
+        public bool IsSectionControlEnabled()
+        {
+            return sectionControlEnabled;
+        }
+
+        private void SetSectionControlEnabled(bool enabled)
+        {
+            if (sectionControlEnabled != enabled)
```

Comment:
Here we could again use an early return:
```suggestion
            if (sectionControlEnabled == enabled)
                return;
```